src/components/DataViewer.vue

```diff
@@ -0,0 +1,92 @@
+<template>
+  <div v-if="datasetIsntEmpty" class="data-viewer-container">
+    <table class="data-viewer-table">
+      <thead class="data-viewer__header">
+        <tr>
+          <td
+            v-for="column in formattedColumns"
+            :class="column.class"
+            :key="column.name"
+            @click="toggleColumnSelection(column.name)"
+          >{{ column.name }}</td>
+        </tr>
+      </thead>
+      <tbody class="data-viewer__body">
+        <tr class="data-viewer__row" v-for="(row, index) in dataset" :key="index">
+          <DataViewerCell
+            v-for="columnName in columnNames"
+            :key="columnName"
+            :isSelected="isSelected(columnName)"
+            :value="getValue(row, columnName)"
+          />
+        </tr>
+      </tbody>
+    </table>
+  </div>
+  <div v-else>No Data Available</div>
+</template>
+<script lang="ts">
+import _ from 'lodash';
+import Vue from 'vue';
+import { Component, Prop } from 'vue-property-decorator';
+import DataViewerCell from './DataViewerCell.vue';
+
+interface DataRow {
+  [key: string]: any;
+}
+
+@Component({
+  name: 'data-viewer',
+  components: {
+    DataViewerCell,
+  },
+})
+export default class DataViewer extends Vue {
+  @Prop({
+    default: () => [],
+    type: Array,
+  })
+  dataset!: Array<object>;
+
+  selectedColumns: Array<string> = [];
+
+  get datasetIsntEmpty() {
+    return this.dataset.length !== 0;
+  }
+
+  get columnNames() {
+    return _.keys(this.dataset[0]);
```

Comment:
I suggest we use standard JS methods when possible:

```suggestion
     return Object.keys(this.dataset[0]);
```